### DISCUSSION TITLE
Briefcase run works dev doesn't: Toga application must have a formal name

### DISCUSSION THREAD

**Discussion by hlovatt** [beeware/toga#3037] [Debugging help]
Hi,

Unusual one:

HelloWorld tutorial App was working on one computer, moved to new MacBook and it stopped working with `briefcase dev` but still works with `briefcase run` (I rebuilt on new machine - didn't try `run` before rebuild). `dev` didn't work before rebuild nor after.

Error is:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<frozen runpy>", line 226, in run_module
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/howardlovatt/repos/beeware/helloworld/src/helloworld/__main__.py", line 4, in <module>
    main().main_loop()
    ^^^^^^
  File "/Users/howardlovatt/repos/beeware/helloworld/src/helloworld/app.py", line 52, in main
    return HelloWorld()
           ^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.12/site-packages/toga/app.py", line 276, in __init__
    raise RuntimeError("Toga application must have a formal name")
RuntimeError: Toga application must have a formal name
```

Have attached log. Note `toml`, it does have a formal name.

Any suggestions?

[briefcase.2024_12_12-13_02_56.dev.log](https://github.com/user-attachments/files/18104684/briefcase.2024_12_12-13_02_56.dev.log)

**Comment by freakboy3742** [comment]
That error indicates that the packaging metadata for the app itself hasn't been written out. There should be a `helloworld.dist-info` directory in your `src` folder (the same location that contains the `helloworld` directory); this is the "formal name" that it's trying to find at runtime (the pyproject.toml is the source of the metadata, not the location that is read at runtime).

That metadata file is written as part of installing the app code. Running `briefcase update` should re-create it in your built app; `briefcase dev -r` should re-create it in your development environment.

You say this has occurred as a result of moving the app to a new computer - how did you do this move? If the `.dist-info` was missing, that *should* have triggered re-creating the file, so I'm not entirely clear how this might have occurred.

**Reply by hlovatt** [reply]
Thanks that has fixed the problem, needed both `update` and `dev -r`.

I moved the files by copying from 1st Mac to iCloud and then from iCloud to 2nd Mac. iCloud only used as a 'dropbox', code not run from there.
